lib/DXIL/DxilNodeProps.cpp

```diff
@@ -81,6 +90,16 @@ bool NodeFlags::GetTrackRWInputSharing() const {
           (uint32_t)DXIL::NodeIOFlags::TrackRWInputSharing) != 0;
 }
 
+void NodeFlags::SetGloballyCoherent() {
+  m_Flags = (DXIL::NodeIOFlags)((uint32_t)m_Flags |
+                                (uint32_t)DXIL::NodeIOFlags::GloballyCoherent);
+}
+
+bool NodeFlags::GetGlobalCoherent() const {
+  return ((uint32_t)m_Flags & (uint32_t)DXIL::NodeIOFlags::GloballyCoherent) !=
```

Comment:
nit: I wonder if `m_Flags` should actually be a `NodeIOFlags`, or if `NodeIOFlags` should be an enum class. One of the benefits of enum classes in some cases is that casting must be explicit, but since the flags will basically always need to be casted it is unfortunate.

Reply:
If I get what you're saying, you're wondering if `m_Flags` should be `uint32_t` instead, since it is `NodeIOFlags` now, (or a regular enum, instead of enum class)?

For the ordinary enum route, we'd need to either extend the value names due to the loss of the enum class namespace or maybe place `NodeIOFlags` under the NodeFlags class, which would also need to be moved to `DxilConstants.h`.  That might not be a bad option.  We would add a `using` statement under the DXIL namespace to import `NodeIOFlags` there, without the enum values polluting the DXIL namespace.

If NodeFlags is moved to `DxilConstants.h`, we'd probably want to inline some/most of the methods as well.  `DxilConstants.h` is intended to define the necessary constants for DXIL production/consumption without having to build/link other DXIL cpp files from DXC, or even depend on LLVM, though it's ok to have some optionally-used functionality declared here and defined elsewhere.

Alternatively, or additionally, we could define global operators for `&` and `|` between types depending whether `NodeIOFlags` is changed and moved under NodeFlags.

In any case, all this seems like something that belongs outside this PR.

Reply:
I'll revisit the flags issue in separate PR, since this one is just adding code with exactly the same pattern, and I'd rather not mix the concern of the best way to solve that with this PR.